pmd-plsql/src/main/java/net/sourceforge/pmd/lang/plsql/ast/PLSQLParser.java

```diff
@@ -4,16 +4,53 @@
 
 package net.sourceforge.pmd.lang.plsql.ast;
 
+import org.checkerframework.checker.nullness.qual.Nullable;
+
 import net.sourceforge.pmd.benchmark.TimeTracker;
 import net.sourceforge.pmd.lang.ast.ParseException;
 import net.sourceforge.pmd.lang.ast.impl.javacc.CharStream;
+import net.sourceforge.pmd.lang.ast.impl.javacc.JavaccToken;
+import net.sourceforge.pmd.lang.ast.impl.javacc.JavaccTokenDocument;
 import net.sourceforge.pmd.lang.ast.impl.javacc.JavaccTokenDocument.TokenDocumentBehavior;
 import net.sourceforge.pmd.lang.ast.impl.javacc.JjtreeParserAdapter;
 import net.sourceforge.pmd.lang.plsql.symboltable.SymbolFacade;
 
 public class PLSQLParser extends JjtreeParserAdapter<ASTInput> {
 
-    private static final TokenDocumentBehavior TOKEN_BEHAVIOR = new TokenDocumentBehavior(PLSQLTokenKinds.TOKEN_NAMES);
+    // Stores images of constant string literals.
+    // This is to reuse the image strings for PLSQL keywords.
+    // JavaCC unfortunately does not store a constant image for those
+    // keywords because the grammar is case-insensitive.
+    // This optimization has the effect that the image of keyword tokens
+    // is always upper-case, regardless of the actual case used in the code.
+    // The original casing can be found by looking at the TextDocument for the file.
+
+    // NOTE: the size of this array should be greater than the number of tokens in the file.
+    private static final String[] STRING_LITERAL_IMAGES_EXTRA = new String[512];
```

Comment:
We seem to have a public static list of all the tokens available:

```suggestion
    private static final String[] STRING_LITERAL_IMAGES_EXTRA = new String[PLSQLTokenKinds.TOKEN_NAMES.size()];
```